### PR TITLE
Procedure Occurrence

### PR DESCRIPTION
**Pull request by csafreen**
This PR transforms data from `intraop.operation`, `postop.renal` and `preop.radiology` tables into `omop_sqldev_schema.procedure_occurrence` table.
Missing:  Procedure Concept id as they are not yet mapped by the vocab team

### CODE REVIEW

**Comment by hamcheezee** [comment]
Since `intra_op__operation` contains many duplicate rows (leading to ~18,000 rows) when joined with the **provider** table, please ensure that the values in important columns are unique. For example, consider using:
```sql
SELECT DISTINCT person_id, procedure_date, procedure_datetime, procedure_end_date, procedure_end_datetime, provider_id, visit_occurrence_id, procedure_source_value
FROM intra_op__operation
```

**Comment by hamcheezee** [comment]
It would be helpful to derive the real `visit_occurrence_id` by mapping the **Visit Occurrence table** using its `session_id`:
```sql
CAST(LEFT(CAST(visit_occurrence_id AS TEXT), LENGTH(CAST(visit_occurrence_id AS TEXT)) - 2) AS INTEGER) AS session_id
``` 

I noticed that `session_id` is missing from **postop.renal** in the spec, which is causing `visit_occurrence_id` to be null. 😂

File: etl/pypasar/db/sql/postgres/procedure_occurrence/stg__procedure_occurrence.sql [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            JOIN {OMOP_SCHEMA}.person AS person ON person.person_source_value = renal.anon_case_no
+        ) AS temp
+    ),
+    pre_op__radiology AS (
+        SELECT ROW_NUMBER() OVER (ORDER BY procedure_date asc) AS procedure_occurence_id, temp.*
+            FROM(
+                SELECT  DISTINCT
+                        person.person_id,
+                        0 AS procedure_concept_id, -- Need concept id mapping
+                        operation_startdate AS procedure_date,
```
[the remark this answers is not in the document]
Comment (csafreen) [reply etl/pypasar/db/sql/postgres/procedure_occurrence/stg__procedure_occurrence.sql:91]: image
According to the mapping in the excel - https://ohdsiorg.sharepoint.com/:x:/r/sites/OHDSIAPAC/_layouts/15/Doc2.aspx?action=edit&sourcedoc=%7B5aa99f90-c21c-4ce4-8edf-70b99dfa0ab4%7D&wdOrigin=TEAMS-MAGLEV.teamsSdk_ns.rwc&wdExp=TEAMS-TREATMENT&wdhostclicktime=1730089132267&web=1, it made sense that procedure date is `operation_startdate`

File: etl/pypasar/db/sql/postgres/procedure_occurrence/stg__procedure_occurrence.sql [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+            temp.provider_id,
+            temp.visit_occurrence_id,
+            temp.visit_detail_id,
+            temp.procedure_source_value,
+            temp.procedure_source_concept_id,
+            temp.modifier_source_value
+        FROM (
+            SELECT  DISTINCT
+                person.person_id,
+                CASE
```
[the remark this answers is not in the document]
Comment (csafreen) [reply etl/pypasar/db/sql/postgres/procedure_occurrence/stg__procedure_occurrence.sql:58]: Sure, but don't have the concept for this one

**Review by hamcheezee — APPROVED** [review]

**Pull request merged**